src/service/routing/osm.clj

```diff
@@ -161,11 +167,13 @@
   only the connected nodes, the points represent the shape of the connection
   and the ways are the metadata associated with the connections"
   [filename]
-  (let [ways          (osm->ways filename)
+  (let [ways          (with-open [file-rdr (bz2-reader filename)]
+                        (osm->ways (:content (xml/parse file-rdr))))
```

Comment:
Can't we encapsulate these lines into a separate fn and reuse it here?

Reply:
yes we could. In fact they were encapsulated before. See the osm->ways declaration from before.
The problem with that approach is that it forces the functions to depend on a certain type of file reader. So if at some point we find out that bz2 compression no longer works, we would be force to modify those functions as well. Even though their logic didnt change at all.

That's why I decided to separate the "where do we get the data" from "how to we process it" ;)